Makefile

```diff
@@ -36,16 +36,27 @@ build: build-assets
 
 HUGO_SERVE := hugo serve --buildDrafts --buildFuture --ignoreVendorPaths="github.com/pulumi/registry/**/*" | grep -v -e 'WARN .* REF_NOT_FOUND'
 
-# bin/resourcedocsgen is considered to depend on all files in the tools/resourcedocsgen directory.
-#
-# Caveat: This does not account for `replace` directives in `go.mod`.
-bin/resourcedocsgen: $(shell find tools/resourcedocsgen | sed 's@ @\\ @g')
+HELPMAKEGO_VERSION := v0.1.0
+HELPMAKEGO := bin/${HELPMAKEGO_VERSION}/helpmakego
+${HELPMAKEGO}:
+	GOBIN=$(shell pwd)/bin/${HELPMAKEGO_VERSION} go install github.com/iwahbe/helpmakego@${HELPMAKEGO_VERSION}
+
+bin:
 	mkdir -p bin
+
+bin/resourcedocsgen: ${HELPMAKEGO} $(shell ${HELPMAKEGO} tools/resourcedocsgen) | bin
```

Comment:
Not a big fan of order-only prereqs .. they make things look kinda complex. The slightly more hacky immediate variable approach we use in providers is a little easier as you don't then need to remember to add `| bin` to all your binary builds:

```
_ := $(shell mkdir -p bin)
```

Reply:
It's also worth doing this for `helpmakego`.